pkg/apis/azureprovider/v1beta1/types.go

```diff
@@ -403,16 +403,26 @@ type Image struct {
 type VMIdentity string
 
 type OSDisk struct {
-	OSType      string                `json:"osType"`
-	ManagedDisk ManagedDiskParameters `json:"managedDisk"`
-	DiskSizeGB  int32                 `json:"diskSizeGB"`
+	OSType           string                `json:"osType"`
+	ManagedDisk      ManagedDiskParameters `json:"managedDisk"`
+	DiskSizeGB       int32                 `json:"diskSizeGB"`
+	DiffDiskSettings *DiffDiskSettings     `json:"diffDiskSettings,omitempty"`
+	// +optional
+	CachingType string `json:"cachingType,omitempty"`
 }
 
 type ManagedDiskParameters struct {
 	StorageAccountType string                       `json:"storageAccountType"`
 	DiskEncryptionSet  *DiskEncryptionSetParameters `json:"diskEncryptionSet,omitempty"`
 }
 
+// DiffDiskSettings describe ephemeral disk settings for the os disk.
+type DiffDiskSettings struct {
```

Comment:
What is `Diff` in this context?

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/docs/book/src/topics/os-disk.md#azure-machine-diffdisksettings